[24.0 backport] seccomp: always allow name_to_handle_at(2)

neersighted opened the pull request:
- Backport of https://github.com/moby/moby/pull/45766 and https://github.com/moby/moby/pull/45837

---

closes #45518

Hi there, this is my first PR here, so please feel free to point me out if anything is wrong with this contribution. I looked the #45518 issue and I believe this could be a fix.

**- What I did**
Removed the function from the filtered syscalls as `name_to_handle_at(2)` is in fact innocuous and safe

**- How I did it**
@neersighted help at [comment](https://github.com/moby/moby/pull/45766#pullrequestreview-1493908145)

**- How to verify it**
N/A

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->
Remove name_to_handle_at(2) from filtered syscalls


**- A picture of a cute animal (not mandatory but encouraged)**
![image](https://github.com/moby/moby/assets/18057391/a8a3b84f-ccb9-40cf-a22c-512713efd790)

bartier committed:
remove name_to_handle_at(2) from filtered syscalls

Signed-off-by: Vitor Anjos <bartier@users.noreply.github.com>
(cherry picked from commit fdc9b7ccebfde394a2c18d25a3d15d27ae72ed53)
Signed-off-by: Bjorn Neergaard <bjorn.neergaard@docker.com>

thaJeztah reviewed (dismissed):
leaving a "request changes" pending outcome of https://github.com/moby/moby/pull/45766#discussion_r1244845677

neersighted committed:
seccomp: add name_to_handle_at to allowlist

Based on the analysis on [the previous PR][1].

  [1]: https://github.com/moby/moby/pull/45766#pullrequestreview-1493908145

Signed-off-by: Bjorn Neergaard <bjorn.neergaard@docker.com>
(cherry picked from commit b335e3d305be86bd28089a057d8be6a346445549)
Signed-off-by: Bjorn Neergaard <bjorn.neergaard@docker.com>

thaJeztah reviewed (approved):
LGTM
